### PR TITLE
(PUP-5123) Replace http:// links with https://

### PR DESCRIPTION
**Pull request by keeleysam**
This replaces most http:// links with https://.  The replaced ones have all been verified to serve identical content.  Those which aren't replaced are contained within XML schema or are internal puppetlabs links which are unable to be verified externally. Some are also outdated and inaccessible and should be updated seperately.

### CODE REVIEW

**Comment by kylog** [comment]
Thanks @keeleysam!

A lot of these are docs site, so ping @nfagerlund.

**Comment by keeleysam** [comment]
It'd be nice if http://links.puppetlabs.com/ was available over a secure connection.

**Comment by nfagerlund** [comment]
I'm checking with our sysops team. I don't think there's any reason we would ever stop serving docs.puppetlabs.com over HTTPS, but I'm gonna make sure! :)

**Comment by danieldreier** [comment]
we already redirect all of puppetlabs.com to HTTPS, sounds fine to me.

File: ext/debian/README.source [hunk 1]
```diff
@@ -1,2 +1,2 @@
 The debian directory is now maintained on Alioth in git. 
-See http://pkg-puppet.alioth.debian.org/ for more information.
+See https://pkg-puppet.alioth.debian.org/ for more information.
```
Comment (ferventcoder) [inline ext/debian/README.source]: Interestingly enough, this looks WAY outdated.

File: ext/debian/copyright [hunk 2]
```diff
@@ -7,7 +7,7 @@ Maintainer:
  Nigel Kersten <nigel@explanatorygap.net>,
  Stig Sandbeck Mathisen <ssm@debian.org>
 Source: git://github.com/puppetlabs/puppet.git,
- http://puppetlabs.com/trac/puppet/wiki/DownloadingPuppet
+ https://puppetlabs.com/trac/puppet/wiki/DownloadingPuppet
```
Comment (ferventcoder) [inline ext/debian/copyright]: This could use an update.

Comment (ferventcoder) [inline ext/debian/copyright]: Perhaps - https://docs.puppetlabs.com/puppet/latest/reference/install_pre.html or http://docs.puppetlabs.com/puppet/latest/reference/install_pre.html#next-install-puppet

File: ext/debian/puppetmaster.README.debian [hunk 3]
```diff
@@ -12,4 +12,4 @@ puppetmaster-passenger package. It will automatically setup an apache2 virtual
 host that will run puppetmaster under Passenger.
 
 For more information:
-http://projects.puppetlabs.com/projects/puppet/wiki/Using_Passenger
+https://projects.puppetlabs.com/projects/puppet/wiki/Using_Passenger
```
Comment (ferventcoder) [inline ext/debian/puppetmaster.README.debian]: This link probably needs a new home.

Comment (ferventcoder) [inline ext/debian/puppetmaster.README.debian]: And then marked as deprecated

File: ext/envpuppet [hunk 4]
```diff
@@ -105,11 +105,11 @@ if [ "${ENVPUPPET_BLEEDING:-}" = "true" ]; then
...
 fi
 
-# http://github.com/jamtur01/puppet-scaffold.git
+# https://github.com/jamtur01/puppet-scaffold.git
 mypath="${mypath}:${ENVPUPPET_BASEDIR}/puppet-scaffold/bin"
 myrubylib="${myrubylib}:${ENVPUPPET_BASEDIR}/puppet-scaffold/lib"
 
-# http://github.com/puppetlabs/puppet-module-tool.git
+# https://github.com/puppetlabs/puppet-module-tool.git
 # Also known as "pmt"  Will become "puppet module"
```
Comment (ferventcoder) [inline ext/envpuppet]: This should be removed entirely.

Comment (ferventcoder) [inline ext/envpuppet]: Lines 114 and 115.

File: ext/ips/puppetagent.xml [hunk 5]
```diff
@@ -34,7 +34,7 @@
       </common_name>
       <documentation>
         <manpage title="puppet" section="1"/>
-        <doc_link name="puppetlabs.com" uri="http://puppetlabs.com/puppet/introduction"/>
+        <doc_link name="puppetlabs.com" uri="https://puppetlabs.com/puppet/introduction"/>
```
Comment (ferventcoder) [inline ext/ips/puppetagent.xml]: This redirects to https://puppetlabs.com/puppet/what-is-puppet - is that where we want the link to go @nfagerlund ?

File: ext/ips/puppetmaster.xml [hunk 6]
```diff
@@ -34,7 +34,7 @@
       </common_name>
       <documentation>
         <manpage title="puppet" section="1"/>
-        <doc_link name="puppetlabs.com" uri="http://puppetlabs.com/puppet/introduction"/>
+        <doc_link name="puppetlabs.com" uri="https://puppetlabs.com/puppet/introduction"/>
```
Comment (ferventcoder) [inline ext/ips/puppetmaster.xml]: Same comment as above. https://puppetlabs.com/puppet/what-is-puppet

File: ext/debian/puppetmaster.README.debian (hunk 3 above)
Comment (ferventcoder) [inline ext/debian/puppetmaster.README.debian]: Apparently that was in another link - https://docs.puppetlabs.com/guides/passenger.html

File: ext/solaris/pkginfo [hunk 7]
```diff
@@ -2,5 +2,5 @@ PKG=CSWpuppet
 NAME=puppet - System Automation Framework
 VERSION=3.0.0
 CATEGORY=application
-VENDOR=http://projects.puppetlabs.com/projects/puppet
+VENDOR=https://projects.puppetlabs.com/projects/puppet
```
Comment (ferventcoder) [inline ext/solaris/pkginfo]: https://tickets.puppetlabs.com/browse/PUP ? Or http://docs.puppetlabs.com/puppet/

File: ext/suse/puppet.spec [hunk 8]
```diff
@@ -9,8 +9,8 @@ Release: 1%{?dist}
 License: Apache 2.0
 Group:    Productivity/Networking/System
 
-URL: http://puppetlabs.com/projects/puppet/
-Source0: http://puppetlabs.com/downloads/puppet/%{name}-%{version}.tar.gz
+URL: https://puppetlabs.com/projects/puppet/
```
Comment (ferventcoder) [inline ext/suse/puppet.spec]: https://tickets.puppetlabs.com/browse/PUP ? Or http://docs.puppetlabs.com/puppet/

File: ext/solaris/smf/puppet.xml [hunk 9]
```diff
@@ -36,7 +36,7 @@
       </common_name>
       <documentation>
         <manpage title="puppet" section="1"/>
-        <doc_link name="puppetlabs.com" uri="http://puppetlabs.com/puppet/introduction"/>
+        <doc_link name="puppetlabs.com" uri="https://puppetlabs.com/puppet/introduction"/>
```
Comment (ferventcoder) [inline ext/solaris/smf/puppet.xml]: Another redirect to  https://puppetlabs.com/puppet/what-is-puppet

File: lib/puppet/parser/functions/hiera.rb [hunk 10]
```diff
@@ -25,7 +25,7 @@ module Puppet::Parser::Functions
   3. Like #1 but with all arguments passed in an array.
 
   More thorough examples of `hiera` are available at:  
-  <http://docs.puppetlabs.com/hiera/1/puppet.html#hiera-lookup-functions>
+  <https://docs.puppetlabs.com/hiera/1/puppet.html#hiera-lookup-functions>
```
Comment (ferventcoder) [inline lib/puppet/parser/functions/hiera.rb]: Probably best to use this link now - https://docs.puppetlabs.com/hiera/3.0/puppet.html#hiera-lookup-functions

Example will need checked though.

File: lib/puppet/parser/functions/hiera_array.rb [hunk 11]
```diff
@@ -27,7 +27,7 @@ module Puppet::Parser::Functions
   If any matched value is a hash, puppet will raise a type mismatch error.
 
   More thorough examples of `hiera` are available at:  
-  <http://docs.puppetlabs.com/hiera/1/puppet.html#hiera-lookup-functions>
+  <https://docs.puppetlabs.com/hiera/1/puppet.html#hiera-lookup-functions>
```
Comment (ferventcoder) [inline lib/puppet/parser/functions/hiera_array.rb]: Probably best to use this link now - https://docs.puppetlabs.com/hiera/3.0/puppet.html#hiera-lookup-functions

Example will need checked though.

File: lib/puppet/parser/functions/hiera_hash.rb [hunk 12]
```diff
@@ -29,7 +29,7 @@ module Puppet::Parser::Functions
   found in the data sources are strings or arrays, puppet will raise a type mismatch error.
 
   More thorough examples of `hiera_hash` are available at:  
-  <http://docs.puppetlabs.com/hiera/1/puppet.html#hiera-lookup-functions>
+  <https://docs.puppetlabs.com/hiera/1/puppet.html#hiera-lookup-functions>
```
Comment (ferventcoder) [inline lib/puppet/parser/functions/hiera_hash.rb]: Probably best to use this link now - https://docs.puppetlabs.com/hiera/3.0/puppet.html#hiera-lookup-functions

Example will need checked though.

File: lib/puppet/parser/functions/hiera_include.rb [hunk 13]
```diff
@@ -39,7 +39,7 @@ module Puppet::Parser::Functions
   3. Like #1 but with all arguments passed in an array.
 
   More thorough examples of `hiera_include` are available at:
-  <http://docs.puppetlabs.com/hiera/1/puppet.html#hiera-lookup-functions>
+  <https://docs.puppetlabs.com/hiera/1/puppet.html#hiera-lookup-functions>
```
Comment (ferventcoder) [inline lib/puppet/parser/functions/hiera_include.rb]: Probably best to use this link now - https://docs.puppetlabs.com/hiera/3.0/puppet.html#hiera-lookup-functions

Example will need checked though.

File: lib/puppet/vendor/deep_merge/deep_merge.gemspec [hunk 14]
```diff
@@ -22,7 +22,7 @@ Gem::Specification.new do |s|
     "lib/deep_merge/rails_compat.rb",
     "test/test_deep_merge.rb"
   ]
-  s.homepage = %q{http://github.com/danielsdeleo/deep_merge}
+  s.homepage = %q{https://github.com/danielsdeleo/deep_merge}
```
Comment (ferventcoder) [inline lib/puppet/vendor/deep_merge/deep_merge.gemspec]: This really did catch everything, didn't it? :D

Comment (keeleysam) [inline lib/puppet/vendor/deep_merge/deep_merge.gemspec]: Absolutely!

File: man/man5/puppet.conf.5 [hunk 15]
```diff
@@ -30,7 +30,7 @@ The Puppet executables will ignore any setting that isn\'t relevant to their fun
 .IP "" 0
 .
 .P
-See the configuration guide \fIhttp://docs\.puppetlabs\.com/guides/configuring\.html\fR for more details\.
+See the configuration guide \fIhttps://docs\.puppetlabs\.com/guides/configuring\.html\fR for more details\.
```
Comment (ferventcoder) [inline man/man5/puppet.conf.5]: This link now points to http://docs.puppetlabs.com/puppet/latest/reference/config_about_settings.html

File: man/man5/puppet.conf.5 [hunk 16]
```diff
@@ -107,7 +107,7 @@ If a custom policy executable is configured, the CA puppet master will run it ev
 If a certificate request is not autosigned, it will persist for review\. An admin user can use the \fBpuppet cert sign\fR command to manually sign it, or can delete the request\.
 .
 .P
-For info on autosign configuration files, see the guide to Puppet\'s config files \fIhttp://docs\.puppetlabs\.com/guides/configuring\.html\fR\.
+For info on autosign configuration files, see the guide to Puppet\'s config files \fIhttps://docs\.puppetlabs\.com/guides/configuring\.html\fR\.
```
Comment (ferventcoder) [inline man/man5/puppet.conf.5]: Auto-sign likely has a new home. A cursory check to the link doesn't show anything.

Comment (ferventcoder) [inline man/man5/puppet.conf.5]: We probably have a much better link now.

File: man/man8/puppet-status.8 [hunk 17]
```diff
@@ -69,7 +69,7 @@ A "true" response or a low\-level connection error\. When used from the Ruby API
 \fBNOTES\fR
 .
 .IP
-This action requires that the server\'s \fBauth\.conf\fR file allow find access to the \fBstatus\fR REST terminus\. Puppet agent does not use this facility, and it is turned off by default\. See \fIhttp://docs\.puppetlabs\.com/guides/rest_auth_conf\.html\fR for more details\.
+This action requires that the server\'s \fBauth\.conf\fR file allow find access to the \fBstatus\fR REST terminus\. Puppet agent does not use this facility, and it is turned off by default\. See \fIhttps://docs\.puppetlabs\.com/guides/rest_auth_conf\.html\fR for more details\.
```
Comment (ferventcoder) [inline man/man8/puppet-status.8]: This now has a home at https://docs.puppetlabs.com/puppet/latest/reference/config_file_auth.html#access-control-behavior

**Comment by ferventcoder** [comment]
Sorry for the impromptu cleanup of links, but a change to https would be a good time to add an additional commit with fixes for the outdated links and examples.

**Comment by ferventcoder** [comment]
Also, please don't use merge - https://github.com/puppetlabs/puppet/blob/master/CONTRIBUTING.md#making-changes (although that doesn't really explain what we hope for when responding to feedback).

If it is necessary for you to get the latest updates, you should always fetch and rebase to master, then rebase your feature branch against the updated master. For reference, this is the workstream we hope for:

* `git fetch upstream` (upstream would be the mainstream repo or `puppetlabs/puppet` in this case)
* `git checkout master`
* `git rebase upstream/master`
* `git checkout your-branch`
* `git rebase master`
* Fix any merge conflicts
* `git push origin your-branch` (origin would be your GitHub repo or `your-github-username/puppet` in this case). You may need to `git push origin your-branch --force` to get the commits pushed. This is generally acceptable with topic branches not in the mainstream repository.

File: ext/debian/control [hunk 18]
```diff
@@ -7,7 +7,7 @@ Build-Depends-Indep: ruby | ruby-interpreter, libopenssl-ruby | libopenssl-ruby1
 Build-Depends: debhelper (>= 7.0.0), openssl
 Standards-Version: 3.9.1
 Vcs-Git: git://github.com/puppetlabs/puppet
-Homepage: http://projects.puppetlabs.com/projects/puppet
+Homepage: https://projects.puppetlabs.com/projects/puppet
```
Comment (ferventcoder) [inline ext/debian/control]: This location needs updated to https://docs.puppetlabs.com/puppet

Comment (ferventcoder) [inline ext/debian/control]: I believe that is correct, @kylog / @nfagerlund can you verify that is where the Homepage should link to?

**Comment by nfagerlund** [comment]
Sysops gives the OK. Go for it!

File: man/man8/puppet-status.8 (hunk 17 above)
Comment (nfagerlund) [inline man/man8/puppet-status.8]: :+1:

File: ext/debian/control (hunk 18 above)
Comment (nfagerlund) [inline ext/debian/control]: Yeah, that old Redmine page may as well go. That's as good a spot as any, I think, unless we want to point to the puppetlabs.com homepage.

File: man/man5/puppet.conf.5 (hunk 16 above)
Comment (nfagerlund) [inline man/man5/puppet.conf.5]: Use https://docs.puppetlabs.com/puppet/latest/reference/ssl_autosign.html.

Comment (nfagerlund) [inline man/man5/puppet.conf.5]: But also, this man page should be automatically generated... I think? I'm not sure what causes it to get updated, anymore, but it's supposed to contain the same content as the setting descriptions in lib/puppet/defaults.rb.

File: lib/puppet/parser/functions/hiera_include.rb (hunk 13 above)
Comment (nfagerlund) [inline lib/puppet/parser/functions/hiera_include.rb]: Use /latest/ instead of /3.0/.

File: lib/puppet/parser/functions/hiera_hash.rb (hunk 12 above)
Comment (nfagerlund) [inline lib/puppet/parser/functions/hiera_hash.rb]: /latest/.

Comment (ferventcoder) [inline lib/puppet/parser/functions/hiera_hash.rb]: thanks!

**Comment by peterhuene** [comment]
HI @keeleysam, thanks for the contribution!  It looks like @ferventcoder had a number of comments where additional changes are required.  With additional fixes, it looks like this is something we would like to merge in.

File: ext/suse/puppet.spec (hunk 8 above)
Comment (ferventcoder) [inline ext/suse/puppet.spec]: @nfagerlund ^^?

Comment (nfagerlund) [inline ext/suse/puppet.spec]: That link still works (via a redirect), and is under Marketing's control. Which I think is as it should be? I'd leave it as is.

Comment (ferventcoder) [inline ext/suse/puppet.spec]: Fair enough.

File: ext/solaris/pkginfo (hunk 7 above)
Comment (ferventcoder) [inline ext/solaris/pkginfo]: Nevermind, leave this one as is.

File: ext/debian/control (hunk 18 above)
Comment (kylog) [inline ext/debian/control]: IIUC the contents of these ext directories is no longer used by PL packaging. Is that correct @melissa or @haus?

**Comment by kylog** [comment]
Hey all, this has tickled a bunch of "holy mackerel, that's an ancient url that may or may not be accurate" comments. Great stuff to audit, but I wonder if we should tackle those issues separately from the intent of this PR, which was to update links from http -> https, not generally rejuvenate the state of external references.

I suggest that because this seems like a Good Thing and docs/ops is cool with it, but it might take a variety of people a bunch of time to run down current links for the various this-and-that urls.

If that seems agreeable, we could ticket the cleanup we've observed and tackle that in separate PRs.

Thoughts?

**Comment by ferventcoder** [comment]
@kylog perhaps. I mentioned something similar in the sea of comments above. :)

**Comment by ferventcoder** [comment]
Mostly not wanting to lose sight of the things that could use fix ups. Prompted based on this change.

File: ext/debian/control (hunk 18 above)
Comment (haus) [inline ext/debian/control]: Yes that is correct.

**Comment by joshcooper** [comment]
Agreed, let's keep this PR focused on changing the URL scheme, and file separate tickets to change content.

@keeleysam it looks like there are now conflicts, could you rebase?

**Comment by joshcooper** [comment]
ping @keeleysam

**Comment by HAIL9000** [comment]
Closing this one in favor of #4433 (which is a rebased version of this pull request)

**Pull request closed without merging**